packages/commerce-sdk-react/src/hooks/ShopperPromotions/query.ts

```diff
@@ -6,24 +6,55 @@
  */
 import {ApiClients, Argument, DataType} from '../types'
 import {useAsync} from '../useAsync'
-import useCommerceApi from '../useCommerceApi'
-import {UseQueryResult} from '@tanstack/react-query'
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperPromotions']
 
+type UsePromotionsParameters = NonNullable<Argument<Client['getPromotions']>>['parameters']
+type UsePromotionsHeaders = NonNullable<Argument<Client['getPromotions']>>['headers']
+type UsePromotionsArg = {
+    headers?: UsePromotionsHeaders
+    rawResponse?: boolean
+} & UsePromotionsParameters
 /**
  * A hook for `ShopperPromotions#getPromotions`.
  * Returns an array of enabled promotions for a list of specified IDs. In the request URL, you can specify up to 50 IDs. If you specify an ID that contains either parentheses or the separator characters, you must URL encode these characters. Each request returns only enabled promotions as the server does not consider promotion qualifiers or schedules.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-promotions?meta=getPromotions} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperpromotions.shopperpromotions-1.html#getpromotions} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const usePromotions = (
-    arg: Argument<Client['getPromotions']>
-): UseQueryResult<DataType<Client['getPromotions']>, Error> => {
-    const {shopperPromotions: client} = useCommerceApi()
-    return useAsync(['promotions', arg], () => client.getPromotions(arg))
+function usePromotions(
+    arg: Omit<UsePromotionsArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['getPromotions']> | Response, Error>
+): UseQueryResult<DataType<Client['getPromotions']>, Error>
+function usePromotions(
+    arg: Omit<UsePromotionsArg, 'rawResponse'> & {rawResponse?: true},
```

Comment:
Minor: We can remove the `?` for rawResponse: true.
Based on the conversation we had in the ShopperSearch PR

Reply:
Good catch, I've removed it.